### PR TITLE
Implement Lava boiler

### PR DESCRIPTION
**Pull request by lynxx131**
This takes the Lava boiler work done a while back by Rundas01 and ports it to the current code base.

This is not yet ready for merging as I still need to figure out a ClassCastException issue.

### CODE REVIEW

File: common/src/main/java/muramasa/gregtech/gui/ScreenLavaBoiler.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    @Override
+    public boolean mouseClicked(double mouseX, double mouseY, int mouseButton) {
+        if ((!AntimatterAPI.isModLoaded(Ref.MOD_JEI) && !AntimatterAPI.isModLoaded(Ref.MOD_REI)) || !container.getTile().has(MachineFlag.RECIPE))
+            return super.mouseClicked(mouseX, mouseY, mouseButton);
+        if (isInGui((imageWidth / 2) - 10, 24, 20, 18, mouseX, mouseY)) {
+            return false;
+        }
+        if (isInGui(115, 43, 18, 18, mouseX, mouseY)) {
```
Comment (Vliro) [inline common/src/main/java/muramasa/gregtech/gui/ScreenLavaBoiler.java]: the gui system does this automatically, it selects the appropriate widget from the gui position

File: common/src/main/java/muramasa/gregtech/gui/widgets/LavaBoilerWidget.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import com.mojang.blaze3d.vertex.PoseStack;
+import muramasa.antimatter.gui.GuiInstance;
+import muramasa.antimatter.gui.IGuiElement;
+import muramasa.antimatter.gui.Widget;
+import muramasa.antimatter.gui.container.ContainerMachine;
+import muramasa.antimatter.gui.widget.WidgetSupplier;
+import muramasa.antimatter.mixin.client.AbstractContainerScreenAccessor;
+import muramasa.gregtech.tile.single.TileEntityLavaBoiler;
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
```
Comment (Vliro) [inline common/src/main/java/muramasa/gregtech/gui/widgets/LavaBoilerWidget.java:12]: in common folder but it uses fabricmc?

File: common/src/main/java/muramasa/gregtech/tile/single/TileEntityLavaBoiler.java [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+
+    public int getMaxHeat() {
+        return recipeHandler.map(r -> (((LavaBoilerRecipeHandler) r).getMaxHeat())).orElse(0);
+    }
+
+    public static class LavaBoilerRecipeHandler extends MachineRecipeHandler<TileEntityLavaBoiler> {
+        int maxHeat = 500, heat, fuel = 0, maxFuel, lossTimer = 0;
+        boolean hadNoWater;
+
+        protected final ContainerData GUI_SYNC_DATA2 = new ContainerData() {
```
Comment (Vliro) [inline common/src/main/java/muramasa/gregtech/tile/single/TileEntityLavaBoiler.java]: do you need this, if you can use the built in gui syncing system?

File: common/src/main/java/muramasa/gregtech/gui/widgets/LavaBoilerWidget.java (hunk 2 above)
Comment (lynxx131) [reply common/src/main/java/muramasa/gregtech/gui/widgets/LavaBoilerWidget.java:12]: This appears to be the case with the existing `CoalBoilerWidget` class which I used as a reference to update the old PR code to fit into the current code base.

Happy to update this and the coal boiler widget if there's a more appropriate way to be doing this.

File: common/src/main/java/muramasa/gregtech/gui/ScreenLavaBoiler.java (hunk 1 above)
Comment (lynxx131) [reply common/src/main/java/muramasa/gregtech/gui/ScreenLavaBoiler.java]: Took this from the existing coal boiler implementation, I'm happy to update as needed but could use a bit of guidance as I've need look at much of the GUI stuff.

Do I just need to remove the above "if" block?

File: common/src/main/java/muramasa/gregtech/tile/single/TileEntityLavaBoiler.java (hunk 3 above)
Comment (lynxx131) [reply common/src/main/java/muramasa/gregtech/tile/single/TileEntityLavaBoiler.java]: As above I used the existing coal boiler as a reference so I'm happy to change as needed, just need a bit of direction or something to use as a reference as I've not done much GUI stuff yet.

File: common/src/main/java/muramasa/gregtech/gui/widgets/LavaBoilerWidget.java (hunk 2 above)
Comment (lynxx131) [reply common/src/main/java/muramasa/gregtech/gui/widgets/LavaBoilerWidget.java:12]: Just checked and Antimatter API also uses the Fabric API `@Environment` annotation in a few places too.

File: common/src/main/java/muramasa/gregtech/gui/ScreenLavaBoiler.java (hunk 1 above)
Comment (lynxx131) [reply common/src/main/java/muramasa/gregtech/gui/ScreenLavaBoiler.java]: Actually I don't think either `ScreenLavaBoiler` or `ScreenCoalBoiler` are actually used any more and should probably be removed.

File: common/src/main/java/muramasa/gregtech/gui/widgets/LavaBoilerWidget.java (hunk 2 above)
Comment (Trinsdar) [reply common/src/main/java/muramasa/gregtech/gui/widgets/LavaBoilerWidget.java:12]: These 2 are fine in common, architectury changes them to the forge version on compile

Comment (lynxx131) [reply common/src/main/java/muramasa/gregtech/gui/widgets/LavaBoilerWidget.java:12]: I'll resolve this based on Trinsdar's comment.

File: common/src/main/java/muramasa/gregtech/gui/ScreenLavaBoiler.java (hunk 1 above)
Comment (lynxx131) [reply common/src/main/java/muramasa/gregtech/gui/ScreenLavaBoiler.java]: I've removed ScreenLavaBoiler as it wasn't doing anything.

I'll remove the ScreenCoalBoiler class in another PR to keep the scope of this one limited.

File: common/src/main/java/muramasa/gregtech/tile/single/TileEntityLavaBoiler.java (hunk 3 above)
Comment (lynxx131) [reply common/src/main/java/muramasa/gregtech/tile/single/TileEntityLavaBoiler.java]: Removed the GUI_SYNC_DATA2 block

**Comment by lynxx131** [comment]
This is all tested and ready for review.

**Comment by lynxx131** [comment]
I'm going to close this PR in favour of a PR that includes both the lava and solar boilers.

**Pull request closed without merging**